Add checklist generation script (#80)
wordpress-mobile/release-toolkit-gutenberg-mobile: Releasing.md

This is useful in case you want to use an unreleased version of the bundle in the apps. For instance, on a PR that's a work in progress, you might want to include to a specific gutenberg-mobile branch in the apps with an updated bundle so reviewers can see the latest changes before approving them (and releasing a new version).

# Release Checklist Script

> Use the './release-checklist.sh' command to generate a release checklist.

Reviewer: ```suggestion
Use the `./release-checklist.sh` command to generate a release checklist.
```

The checklist will be published as an issue on the [Gutenberg Mobile GitHub repository](https://github.com/wordpress-mobile/gutenberg-mobile/issues).

> The script will generate a checklist for each of the [release types](#different-types-of-releases).

Reviewer: The release types in this document are:
- Alpha
- Regular
- Betafix
- Hotfix

However, the release checklist script expects the release type values:
- `scheduled`
- `beta`
- `hotfix`

I'm wondering if we should either add alternative names (like `scheduled` for the regular releases) to this document or use the same names to prevent misleading, wdyt?

> With out any arguemnts, the script will prompt for all the checklist options. You can use command line options to skip the prompts. To see the full list of options, run

Reviewer: ```suggestion
Without any arguments, the script will prompt for all the checklist options. You can use command line options to skip the prompts. To see the full list of options, run:
```

```
./release-checklist.sh -h
```

The script can also be used to update an existing checklist to add:

1. Aztec update steps
2. Incoming changes

Use the `-i` option with the id of the issue to use the update feature. Then include either the `-a` or `-u` options to add or update the checklist with Aztec or incoming changes steps, respectively.

# Release Automation Script

The `./release-automation.sh` script is used to automate the release process.
It will create 4 branches/PRs:

1. `release/X.XX.X` on [mobile-gutenberg](https://github.com/wordpress-mobile/gutenberg-mobile)
2. `rnmobile/release_X.XX.X` on [wordpress/gutenberg](https://github.com/WordPress/gutenberg)

> 3. `gutenberg/integrate_release_X.X.X` on [wordpress-mobile/WordPress-Android](https://github.com/wordpress-mobile/WordPress-Android

Reviewer: ```suggestion
3. `gutenberg/integrate_release_X.X.X` on [wordpress-mobile/WordPress-Android](https://github.com/wordpress-mobile/WordPress-Android)
```

1. After the fix is merged and if there is an ongoing regular release, betafix or hotfix then the changes should be brought back to those branches as well.

Rest should be same the as Betafix

# Handling Aztec Updates

> Before creating the release, make sure Aztec is up to date. Verify `[gutenberg-mobile/RNTAztecView.podspec](#)` and `gutenberg-mobile/gutenberg/packages/react-native-aztec/RNTAztecView.podspec` refer to the same `Wordpress-Aztex-iOS` version and are poiting to a stable tagged release (e.g. 1.14.1). If not, we may need to create a new Aztec release.

Reviewer: ```suggestion
Before creating the release, make sure Aztec is up to date. Verify `[gutenberg-mobile/RNTAztecView.podspec](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RNTAztecView.podspec)` and [`gutenberg-mobile/gutenberg/packages/react-native-aztec/RNTAztecView.podspec`](https://github.com/WordPress/gutenberg/blob/trunk/packages/react-native-aztec/RNTAztecView.podspec) refer to the same `WordPress-Aztec-iOS` version and are poiting to a stable tagged release (e.g. 1.14.1). If not, we may need to create a new Aztec release.
```

The `./release_automation.sh` will verify if the Aztec version is correct. If not run `./release_checklist.sh -a -i {release issue id}` to update the release checklist.

### Updating Aztec iOS

Follow [this process](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Documentation/ReleaseProcess.md) for iOS. Afterwards update Aztec version references within `gutenberg-mobile/RNTAztecView.podspec` and `gutenberg-mobile/gutenberg/packages/react-native-aztec/RNTAztecView.podspec` to the new WordPress-Aztec-iOS version.

### Updating Aztec Android

> For Android, release are created via releases. Go to [AztecEditor-Android/releases](https://github.com/wordpress-mobile/AztecEditor-Android/releases) and draft a new release with the new version tag. Use the version for the release title and add the changelog as the descriptions. The binaries are created when the release is published.

Reviewer: ```suggestion
For Android, release are created via GitHub releases. Go to [AztecEditor-Android/releases](https://github.com/wordpress-mobile/AztecEditor-Android/releases) and draft a new release with the new version tag. Use the version for the release title and add the changelog as the description. The binaries are created when the release is published.
```

# Handling Incoming Changes

> If addiotional changes (e.g. bug fixes) were merged into `gutenberg-mobile/release/X.XX.X` or `gutenberg/rnmobile/release-X.XX.X` branches add an incoming changes checklist to the release issue. The addtion can be added by running:

Reviewer: ```suggestion
If additional changes (e.g. bug fixes) were merged into `gutenberg-mobile/release/X.XX.X` or `gutenberg/rnmobile/release-X.XX.X` branches add an incoming changes checklist to the release issue. The addition can be added by running:
```

```

>  `./release_checklist.sh -u -i {release issue id}` -m "Some info about the incoming change".

Reviewer: ```suggestion
./release_checklist.sh -u -i {release issue id} -m "Some info about the incoming change".
```

```

# Verify Localization Strings

> Bundling will generate two location strings files:

Reviewer: ```suggestion
Bundling will generate two localization strings files:
```

- [bundle/android/strings.xml](https://href.li/?https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/bundle/android/strings.xml)
- [bundle/ios/GutenbergNativeTranslations.swift](https://href.li/?https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/bundle/ios/GutenbergNativeTranslations.swift)

> These files will only be modified if there are any string changes. Any modfidications to these files should be verified for correctness:

Reviewer: ```suggestion
These files will only be modified if there are any string changes. Any modifications to these files should be verified for correctness:
```
